src/internal/ResourceCall.js

```diff
@@ -8,36 +7,67 @@ class ResourceCall extends Smart {
   defaults(context) { return { context }; }
 
   _exceptionParser(exception) {
-    if (exception.statusCode && exception.response) {
+    if (exception.response) {
       return {
-        message: exception.response.body.message || exception.name,
-        code: exception.response.body.code || exception.statusCode,
-        status: exception.response.body.status || _.snakeCase(exception.response.statusMessage)
+        message: exception.message,
+        code: exception.response.status,
+        status: exception.response.statusText,
       };
     }
     return exception;
   }
 
+  _fetchResponseParser(response) {
+    const { responseType } = this.context;
+    switch (responseType.toLowerCase()) {
+      case 'json':
+        return response.json();
+
+      case 'text':
+        return response.text();
+
+      case 'formdata':
+        return response.formData();
+
+      case 'arraybuffer':
+        return response.arrayBuffer();
+
+      case 'blob':
+        return response.blob();
+
+      default:
+        throw new Error('Unsupported response type');
+    }
+  }
+
+  _promiseAllObject(obj) {
```

Comment:
`_awaitAllObject()` ? 
and why is this not used anywhere ?

Reply:
forgot to remove it !